### DISCUSSION TITLE
Build for ARM

### DISCUSSION THREAD

**Discussion by Holandsoest** [DBraun/DawDreamer#168] [Ideas]
I am trying to use this module for my python project, we are in testing fase, but found out it does not work on our Nividea Jetson Nano ( ARM ).  
`pip install dawdreamer` yields:  
```
ERROR: Could not find a version that satisfies the requirements dawdreamer
ERROR: No matching distribution found for dawdreamer
```

*If you know workarounds. Please share*

**Comment by DBraun** [comment]
Correct, there aren't ARM Linux builds on PyPI right now. However, you should be able to build DawDreamer locally for ARM linux, as long as you remove the BUILD_DAWDREAMER_FAUST "Extra Preprocessor Definition" from the projucer file.

![image](https://github.com/DBraun/DawDreamer/assets/2096055/4589cc90-6896-45a4-a01d-c481986df649)

Then you would follow the build instruction in the developer's guide: https://github.com/DBraun/DawDreamer/wiki/Developer's-Guide

I'll look into whether I can put this kind of build (without libfaust) on PyPI in the short term. Supporting libfaust additionally would be a few more steps.

**Reply by Holandsoest** [reply]
Thank you so much for responding @DBraun,
I successfully build the wheel for arm64 for my Ubuntu Jetson nano :tada: 

I am a junior dev and it took me some time to understand what you meant. :)
I tried *(since i don't always have access to the target)*  your solution out on my `x86` vmware ubuntu_desktop and it worked. I was able to build a wheel that [did not pass all the tests](https://github.com/Holandsoest/art-to-music/issues/26), but it made sense since i disabled `BUILD_DAWDREAMER_FAUST`

## Notes: 
I have a few notes on where i made some mistakes. I will place them here in case other people from the interwebs also have the same issue.

1. You will need to build [Juce](https://github.com/juce-framework/JUCE) for arm
by default it is not build for arm so `sudo apt install juce-tools`, did not work for me. *since that is a old verison or some other unknown magic*
Instead you want to `git clone https://github.com/juce-framework/JUCE.git` somewhere. I did it in my downloads folder.
go to `~/Downloads/JUCE/extras/Projucer/Builds/LinuxMakefile` and then run `make`
now you get the Projucer app, but you need to cp it into your `/usr/bin/` with: `cp `wherever the Projucer is` /usr/bin/Projucer`
Now if you run from your home `Projucer` it will start up

2. Follow @DBraun directive above, but in `Projucer /DawDreamer/DawDreamer.jucer` you also need to remove the `faustwithllvm` from `External Libraries to Link` for Linux Makefile  
![Screenshot from 2023-06-06 12-46-38](https://github.com/DBraun/DawDreamer/assets/103452406/81a74e35-ef5c-45ec-89a8-adf2ef1171a0)


3. and also i ran into a missing python.h error, but after i added the following on the top of my `/DawDreamer/build_linux.sh` it was fine. 
```
PYTHONLIBPATH="/usr/lib/python3.10"
PYTHONINCLUDEPATH="/usr/include/python3.10"
```
What i think happened is:
You need to run  `sudo sh build_linux.sh`, because `sudo` is required for `apt` and some other, but because it is `sudo` my external set variables where gone???
So when you define it in the top of the sh script it is fine!
